src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/ProjectSystem/DocumentContextExtensions.cs

```diff
@@ -27,13 +33,25 @@ public static async Task<Document> GetGeneratedDocumentAsync(this VersionedDocum
         var projectKey = razorDocument.Project.ToProjectKey();
         var generatedFilePath = filePathService.GetRazorCSharpFilePath(projectKey, razorDocument.FilePath.AssumeNotNull());
         var generatedDocumentId = solution.GetDocumentIdsWithFilePath(generatedFilePath).First(d => d.ProjectId == razorDocument.Project.Id);
-        var generatedDocument = solution.GetDocument(generatedDocumentId).AssumeNotNull();
+        generatedDocument = solution.GetDocument(generatedDocumentId).AssumeNotNull();
```

Comment:
We have a `GetRequiredDocument(DocumentId)` extension method for `Project`. Should we also have one for `Solution`?

Reply:
May as well.